perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_def_maxUserDataQuotas.java

```diff
@@ -0,0 +1,50 @@
+package cz.metacentrum.perun.core.impl.modules.attributes;
+
+import cz.metacentrum.perun.core.api.Attribute;
+import cz.metacentrum.perun.core.api.AttributeDefinition;
+import cz.metacentrum.perun.core.api.AttributesManager;
+import cz.metacentrum.perun.core.api.Resource;
+import cz.metacentrum.perun.core.api.exceptions.InternalErrorException;
+import cz.metacentrum.perun.core.api.exceptions.WrongAttributeAssignmentException;
+import cz.metacentrum.perun.core.api.exceptions.WrongAttributeValueException;
+import cz.metacentrum.perun.core.api.exceptions.WrongReferenceAttributeValueException;
+import cz.metacentrum.perun.core.impl.PerunSessionImpl;
+import cz.metacentrum.perun.core.implApi.modules.attributes.ResourceAttributesModuleAbstract;
+import cz.metacentrum.perun.core.implApi.modules.attributes.ResourceAttributesModuleImplApi;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Attribute for setting maximum of data quotas for any volume on defined resource.
+ * By this attribute, facility manager is able to set maximum for defined resource
+ * which must be adhere by Vo Manager of assigned Vo.
+ *
+ * @author Michal Stava stavamichal@gmail.com
+ */
+public class urn_perun_resource_attribute_def_def_maxUserDataQuotas extends ResourceAttributesModuleAbstract implements ResourceAttributesModuleImplApi {
+
+	@Override
+	public void checkAttributeValue(PerunSessionImpl perunSession, Resource resource, Attribute attribute) throws InternalErrorException, WrongAttributeValueException, WrongReferenceAttributeValueException, WrongAttributeAssignmentException {
+		//attribute can be null, it means there are no max user settings on resource
+		if(attribute.getValue() == null) {
+			return;
+		}
+
+		//Check if every part of this map has the right pattern
+		//And also check if every quota part has right settings (softQuota<=hardQuota)
+		perunSession.getPerunBl().getModulesUtilsBl().checkAndTransferQuotas(attribute, resource, null, true);
+	}
+
+	@Override
+	public AttributeDefinition getAttributeDefinition() {
+		AttributeDefinition attr = new AttributeDefinition();
+		attr.setNamespace(AttributesManager.NS_RESOURCE_ATTR_DEF);
+		attr.setFriendlyName("maxUserDataQuotas");
+		attr.setDisplayName("Maximum of data quotas of User on any volumes.");
+		attr.setType(LinkedHashMap.class.getName());
+		attr.setDescription("This is Facility manager maximum set for Resource on defined volumes. " +
```

Comment:
I would replace beginning of the text with "Maximum data quota for each user on this resource."

"Facility manager maximum" don't have to be clear for everyone. Also, I wouldn't mentioned explicit role here, because that might change.

Reply:
Fixed.